### ISSUE TITLE
Suggestion: incorporate depth supervision into #2521

### ISSUE THREAD

**Issue by machenmusik** [nerfstudio-project/nerfstudio#2572]
Per side discussion, incorporating depth supervision into #2521 may help some issues encountered; masks could be represented as infinite depth as an initial degenerate case.

**Comment by nikmo33** [comment]
@machenmusik This degenerate case will only work if the mask is used to mask out the background in an object centric scene? For example in a scene centric scene, masks can often represent dynamic objects - eg. humans, moving vehicles etc. and the proposed degenerate case would cause incorrect depth supervision on these points.

**Comment by machenmusik** [comment]
@nikmo33 agreed, that was the subject of prior discussion - would want that behavior to be selectable but off by default

**Comment by machenmusik** [comment]
(For the degenerate case, I am not sure whether #2165 works in the object centric usage of #2521; if so, no need)
